plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -179,29 +224,28 @@ func (m *MetricsReader) executeQuery(ctx context.Context, p metricsQueryParams)
 		Step:  *p.Step,
 	}
 
-	m.logger.Debug("Executing Prometheus query", zap.String("query", promQuery), zap.Any("range", queryRange))
-
 	mv, warnings, err := m.client.QueryRange(ctx, promQuery, queryRange)
 	if err != nil {
 		logErrorToSpan(span, err)
 		return &metrics.MetricFamily{}, fmt.Errorf("failed executing metrics query: %w", err)
 	}
 	if len(warnings) > 0 {
-		m.logger.Warn("Warnings detected on Prometheus query", zap.Any("warnings", warnings))
+		m.logger.Warn("Warnings detected on Prometheus query", zap.Any("warnings", warnings), zap.String("query", promQuery), zap.Any("range", queryRange))
 	}
 
-	m.logger.Debug("Prometheus query results", zap.String("results", mv.String()))
-	return dbmodel.ToDomainMetricsFamily(
+	m.logger.Debug("Prometheus query results", zap.String("results", mv.String()), zap.String("query", promQuery), zap.Any("range", queryRange))
```

Comment:
It was hard to correlate the request with the response, so I just decided to move the query string in the response, which also helps reduce noise in logs.